a11y-dialog.js

```diff
@@ -1,148 +1,308 @@
+    if (typeof target === 'string') {
+      return $$(target);
+    }
   }
 
-  // Helper function to get all focusable children from a node
-  function getFocusableChildren (node) {
-    var focusableElements = ['a[href]', 'area[href]', 'input:not([disabled])', 'select:not([disabled])', 'textarea:not([disabled])', 'button:not([disabled])', 'iframe', 'object', 'embed', '[contenteditable]', '[tabindex]:not([tabindex^="-"])'];
+  // Set the focus to the first focusable child of the given element
+  // @param {Element} node
+  function setFocusToFirstItem (node) {
+    var focusableChildren = getFocusableChildren(node);
 
-    return $$(focusableElements.join(','), node).filter(function (child) {
-      return !!(child.offsetWidth || child.offsetHeight || child.getClientRects().length);
-    });
+    if (focusableChildren.length) {
+      focusableChildren[0].focus();
+    }
   }
 
-  // Helper function to get all nodes in context matching selector as an array
-  function $$ (selector, context) {
-    return Array.prototype.slice.call((context || document).querySelectorAll(selector));
+  // Get the focusable children of the given element
+  // @param {Element} node
+  // @return {Array<Element>}
+  function getFocusableChildren (node) {
+    return $$(FOCUSABLE_ELEMENTS.join(','), node).filter(function (child) {
+      return !!(child.offsetWidth || child.offsetHeight || child.getClientRects().length);
+    });
   }
 
-  // Helper function trapping the tab key inside a node
+  // Trap the focus inside the given element
+  // @param {Element} node
+  // @param {Event} event
   function trapTabKey (node, event) {
     var focusableChildren = getFocusableChildren(node);
     var focusedItemIndex = focusableChildren.indexOf(document.activeElement);
 
+    // If the SHIFT key is being pressed while tabbing (moving backwards) and
+    // the currently focused item is the first one, move the focus to the last
+    // focusable item from the dialog element
     if (event.shiftKey && focusedItemIndex === 0) {
       focusableChildren[focusableChildren.length - 1].focus();
       event.preventDefault();
+    // If the SHIFT key is not being pressed (moving forwards) and the currently
+    // focused item is the last one, move the focus to the first focusable item
+    // from the dialog element
     } else if (!event.shiftKey && focusedItemIndex === focusableChildren.length - 1) {
       focusableChildren[0].focus();
       event.preventDefault();
     }
   }
 
-  // Helper function to focus first focusable item in node
-  function setFocusToFirstItem (node) {
-    var focusableChildren = getFocusableChildren(node);
-    if (focusableChildren.length) focusableChildren[0].focus();
+  // Retrieve siblings from given element
+  // @param {Element} node
+  // @return {Array<Element>}
+  function getSiblings (node) {
+    var siblings = toArray(node.parentNode.childNodes)
+    siblings.splice(siblings.indexOf(node), 1)
+    return siblings
```

Comment:
Actually you might have been commenting on a commit rather than the PR code itself, because this has been fixed already.